tests/components/unifi/test_controller.py

```diff
@@ -293,33 +321,132 @@ async def test_reset_after_successful_setup(hass, aioclient_mock):
     assert len(controller.listeners) == 0
 
 
+async def test_reset_fails(hass, aioclient_mock):
+    """Calling reset when the entry has been setup can return false."""
+    config_entry = await setup_unifi_integration(hass, aioclient_mock)
+    controller = hass.data[UNIFI_DOMAIN][config_entry.entry_id]
+
+    with patch(
+        "homeassistant.config_entries.ConfigEntries.async_forward_entry_unload",
+        return_value=False,
+    ):
+        result = await controller.async_reset()
+        await hass.async_block_till_done()
+
+    assert result is False
+
+
+async def test_connection_state_signalling(hass, aioclient_mock, mock_unifi_websocket):
+    """Verify signalling and connection state are properly represented."""
+    config_entry = await setup_unifi_integration(hass, aioclient_mock)
+    controller = hass.data[UNIFI_DOMAIN][config_entry.entry_id]
+
+    event_call = Mock()
+    unsub = async_dispatcher_connect(hass, controller.signal_reachable, event_call)
+
+    assert controller.available is True
```

Comment:
Can we assert the state of a unifi entity in the state machine instead? The available property of the client entity class seems to correspond to the controller available property.

Reply:
Yes that should work

Reply:
Fixed